packages/calcite-components/src/components/action-menu/action-menu.tsx

```diff
@@ -401,7 +401,21 @@ export class ActionMenu implements LoadableComponent {
       .assignedElements({
         flatten: true,
       })
-      .filter((el) => el?.matches("calcite-action")) as HTMLCalciteActionElement[];
+      .reduce((previousValue: HTMLCalciteActionElement[], currentValue) => {
+        if (currentValue?.matches("calcite-action")) {
+          return [...previousValue, currentValue];
+        }
+
+        if (currentValue?.matches("calcite-action-group")) {
+          const groupActions = Array.from(
+            currentValue.querySelectorAll("calcite-action")
+          ) as HTMLCalciteActionElement[];
+
+          return [...previousValue, ...groupActions];
```

Comment:
This can be `return previousValue.push(...groupActions)` for better performance

Reply:
done!

Reply:
> for better performance

Just my two cents: unless we're dealing with a critical path, prioritizing readability/maintainability over micro-optimizations might be more beneficial. It's easy to get caught in the perf optimization rabbit hole with little to gain. Also, it's worth noting that modern browsers are pretty good at optimizing code themselves.